extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -699,7 +698,7 @@ public static Table readPartitionedTableInferSchema(
         }
         // TODO (https://github.com/deephaven/deephaven-core/issues/877): Support schema merge when discovering multiple
         // parquet files
-        final ParquetTableLocationKey firstKey = foundKeys.get(0);
+        final ParquetTableLocationKey firstKey = sortedKeys.getKnownKeys().get(0);
```

Comment:
It would be best to use the *last* key, not the first. We should also update the JavaDoc:
"using a definition built from the highest (by {@link ParquetTableLocationKey location key} order) location found"

In general, locations are sorted such that the most recent comes last; using the last one to define the definition for the data set makes sense. The previous approach allowed the key finder to determine which one was most important, but this approach isn't unreasonable.

Reply:
I thought about that, but shouldn't we consider that a breaking change? Happy to make it, just wanting to make sure I'm thinking about it correctly.

Do any of the existing key finders apply any specific orderings to the keys?

Reply:
1. I don't think we should consider it a breaking change; I bet 99.9% of our users have never seen that JavaDoc, and are using us on datasets where all files have the same schema. Anyone who is using us "in production" should probably be specifying a definition or using metadata files.
2. In all cases but one, they supply locations in directory traversal order. The exception is the metadata file-based key finder, which will use the order found in the `_metadata` file.
3. Sorting on the TLKs is applied *later* when accepting non-empty locations to include in the table row set, and that order is strictly defined.